### PR TITLE
Ability to upload additional documents! 📖 

### PR DESCRIPTION
**Pull request by sdevalapurkar-bcgov**
# Description

This PR includes the following proposed change(s):

- ability to upload additional documents!! 📖 
- JIRA: https://justice.gov.bc.ca/jira/browse/FLA-192. 

Specifically:
JIRAs: https://justice.gov.bc.ca/jira/browse/FLA-359, https://justice.gov.bc.ca/jira/browse/FLA-360, and https://justice.gov.bc.ca/jira/browse/FLA-363

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

- locally
- jest

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: src/frontend/efiling-frontend/src/components/page/upload/Upload.js [hunk 1]
```diff
@@ -17,74 +18,192 @@ import { propTypes } from "../../../types/propTypes";
...
+  const isValid = (currentValue) =>
+    Object.prototype.hasOwnProperty.call(currentValue, "isAmendment") &&
+    Object.prototype.hasOwnProperty.call(
+      currentValue,
+      "isSupremeCourtScheduling"
+    );
+
+  if (filesToUpload.documents.length === 0) setContinueBtnEnabled(false);
+  else if (filesToUpload.documents.every(isValid)) setContinueBtnEnabled(true);
+};
```
Comment (adodge-bcg) [inline src/frontend/efiling-frontend/src/components/page/upload/Upload.js:35]: Nit: Should this be setting the setContinueBtnEnabled or just returning a true or false?

File: src/frontend/efiling-frontend/src/components/page/upload/Upload.js [hunk 2]
```diff
@@ -17,74 +18,192 @@ import { propTypes } from "../../../types/propTypes";
...
+
+const generateDropdownItems = ({ level, courtClass }, setItems, items) => {
+  if (items.length > 0) return;
+
+  axios
+    .get(`/lookup/documentTypes/${level}/${courtClass}`)
+    .then(({ data: { documentTypes } }) => setItems(documentTypes))
+    .catch(() => window.open(sessionStorage.getItem("errorUrl"), "_self"));
+};
```
Comment (adodge-bcg) [inline src/frontend/efiling-frontend/src/components/page/upload/Upload.js:44]: Same here, calling setItems internally or just returning the documentTypes list?

Comment (sdevalapurkar-bcgov) [reply src/frontend/efiling-frontend/src/components/page/upload/Upload.js:44]: renamed to `setDropdownItems`

File: src/frontend/efiling-frontend/src/components/page/upload/Upload.js (hunk 1 above)
Comment (sdevalapurkar-bcgov) [reply src/frontend/efiling-frontend/src/components/page/upload/Upload.js:35]: awesome suggestion!

**Pull request merged**